server/game/player.js

```diff
@@ -1220,23 +1220,24 @@ class Player extends Spectator {
     // If no callback is passed, pulled card is returned, but if it is joker the 
     // value selection if needed has to be handled by the caller.
     // The pulled card has to be taken care of manually afterwards.
-    pull(callback, addMessage = false) {
+    pull(callback, addMessage = false, props = {}) {
         if(this.drawDeck.length === 0) {
             this.shuffleDiscardToDrawDeck();
         }
         let pulledCard = this.drawDeck[0];
         this.moveCard(pulledCard, 'being played');
-        this.game.raiseEvent('onCardPulled', { card: pulledCard });
         if(addMessage) {
             this.game.addMessage('{0} pulled {1}of{2}({3} )', this, pulledCard.value, pulledCard.suit, pulledCard);
         }
-        if(callback) {
-            if(pulledCard.getType() === 'joker') {
-                this.game.queueStep(new JokerPrompt(this.game, pulledCard, callback));
-            } else {
-                callback(pulledCard, pulledCard.value, pulledCard.suit);
+        this.game.raiseEvent('onCardPulled', { card: pulledCard, props }, event => {
+            if(callback) {
+                if(event.card.getType() === 'joker') {
+                    this.game.queueStep(new JokerPrompt(this.game, event.card, callback));
+                } else {
+                    callback(event.card, event.card.value, event.card.suit);
+                }
```

Comment:
I put the whole logic that should be executed after pull (callback) inside the event handler so it can be replaced by the Emre's ability. It will keep the pull event handler the same, but will feed a different pulledCard into the callback.